// express-methods/dist/app.util.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var httpMethod;
(function (httpMethod) {
    httpMethod["GET"] = "GET";
})(httpMethod = exports.httpMethod || (exports.httpMethod = {}));
var ExpressError = /** @class */ (function (_super) {
    __extends(ExpressError, _super);
    function ExpressError(obj, code, verbose) {
        if (code === void 0) { code = 400; }
        if (verbose === void 0) { verbose = {}; }
        var _this = _super.call(this, obj) || this;
        _this.statusCode = null;
        _this.verbose = {};
        _this.statusCode = code;
        _this.verbose = verbose;
        return _this;
    }
    return ExpressError;
}(Error));
exports.ExpressError = ExpressError;
exports.attachFinishMethod = function (req, res, next) {
    var defaultHttpStatus = 200;
    if (req.method !== httpMethod.GET) {
        defaultHttpStatus = 201;
    }
    res.finish = function (data, message, httpStatus, errors) {
        if (data === void 0) { data = {}; }
        if (message === void 0) { message = ""; }
        if (httpStatus === void 0) { httpStatus = defaultHttpStatus; }
        if (errors === void 0) { errors = {}; }
        res.status(httpStatus).send({
            message: message,
            data: data,
            errors: errors
        });
    };
    next();
};
exports.expressErrorHandler = function (err, req, res, next) {
    var _a = err.statusCode, statusCode = _a === void 0 ? 400 : _a, verbose = err.verbose;
    res.finish({}, err.message, statusCode, verbose);
};
